src/Config/ConfigOption.php

```diff
@@ -0,0 +1,290 @@
+                    return $default;
+                }
+                return $fn_format_loaded_value($user->fields[$this->getName()]) ?? $default;
+        }
+        return $default;
+    }
+
+    /**
+     * Set the value of the configuration option for a specific scope.
+     * @param ConfigScope $scope The scope to set the value for
+     * @param mixed $value The value to set
+     * @param array $scope_params The parameters to use when looking up where to set the value based on the scope (e.g. Entity ID)
+     * @return void
+     */
+    public function setValue(ConfigScope $scope, mixed $value, array $scope_params = []): void
+    {
+        if (!in_array($scope, $this->scopes, true)) {
+            return;
+        }
+        if (is_array($value)) {
+            $value = exportArrayToDB($value);
+        }
+        switch ($scope) {
+            case ConfigScope::GLOBAL:
+                \Config::setConfigurationValues($scope_params['context'] ?? 'core', [$this->getName() => $value]);
+                break;
+            case ConfigScope::ENTITY:
+                $entity = new \Entity();
+                if ($entity->getFromDB($scope_params['entities_id'] ?? \Session::getActiveEntity())) {
+                    $entity->update([
+                        $this->getName() => $value,
+                    ]);
+                }
+                break;
+            case ConfigScope::USER:
+                $user = new \User();
+                if ($user->getFromDB($scope_params['users_id'] ?? \Session::getLoginUserID())) {
+                    $user->update([
+                        'id' => $user->getID(),
+                        $this->getName() => $value,
+                    ]);
+                }
+                break;
+        }
+    }
+
+    /**
+     * Get the computed value of the configuration option for the given scope.
+     * For example, if this option exists in the {@link ConfigScope::GLOBAL} and {@link ConfigScope::USER} scopes, and
+     * is set at the global scope but not the user scope, this will return the global value. If it is set at both
+     * scopes, the user value will be returned.
+     * The order of precedence is determined by the weight of the scope.
+     * @param $scope_params
+     * @param mixed|null $default
+     * @return mixed
+     * @see ConfigScope::getWeight()
+     */
+    public function getComputedValue($scope_params = [], mixed $default = null): mixed
+    {
+        $scopes = $this->getScopes();
+        // Sort by highest weight first. We will work from most specific to least specific until we find a value.
+        usort($scopes, static fn(ConfigScope $a, ConfigScope $b) => $b->getWeight() <=> $a->getWeight());
+        foreach ($scopes as $scope) {
+            if ($scope === ConfigScope::ENTITY) {
+                return Entity::getUsedConfig(
+                    fieldref: $scope_params['reference_field'] ?? $this->getName(),
```

Comment:
I think that the `reference_field` must be defined in the `ConfigOption` itself. Indeed, to get the value for a given field, we should always use the same `reference_field`. The developer that call `getComputedValue()` should not have to specify this value, to avoid mistakes.